src/pool/srv_target.c

```diff
@@ -2438,8 +2438,13 @@ cont_discard_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 
 put:
 	ds_cont_child_put(cont);
+	/* don't destroy vos container, to avoid ds_cont_tgt_refresh_agg_eph() failure,
+	 * later depend on container recovery process to handle it.
+	 */
+#if 0
 	if (rc == 0)
 		rc = ds_cont_child_destroy(arg->tgt_discard->pool_uuid, entry->ie_couuid);
+#endif
```

Comment:
Some confused. Before incremental reintegration being landed, if without discarding the old container, how can we guarantee the old/stale (punched) object/key can be removed after reintegration?

Reply:
all object already discarded in obj_discard_cb(), here just keep the vos container now.